app/src/main/java/com/sdp13epfl2021/projmag/database/impl/firebase/FirebaseProjectDatabase.kt

```diff
@@ -151,10 +151,15 @@ class FirebaseProjectDatabase(private val firestore: FirebaseFirestore) : Projec
         onFailure: (Exception) -> Unit
     ) {
         project.let {
-            firestore.collection(ROOT).add(
-                it.toMapString()
-            )
-                .addOnSuccessListener { id -> onSuccess(id.id) }
+            if (project.id.isEmpty()) {
+                firestore.collection(ROOT).add(
+                    it.toMapString()
+                ).addOnSuccessListener { id -> onSuccess(id.id) }
```

Comment:
The lambda name "id" was missnamed, because it is a DocumentReference.
```suggestion
                ).addOnSuccessListener { doc -> onSuccess(doc.id) }
```